src/rgw/rgw_process.cc

```diff
@@ -312,6 +313,22 @@ int process_request(rgw::sal::RGWRadosStore* const store,
     handler->put_op(op);
   rest->put_handler(handler);
 
+  // access log line elements begin per Apache Combined Log Format with additions following
+  static bool beast_framework = s->cct->_conf->rgw_frontends.find("beast") != string::npos;
+  if ( beast_framework ) {
+    std::stringstream buf;
+    bool user_agent_hdr = rgw_env.get("HTTP_USER_AGENT") != nullptr;
+    bool referer_hdr = rgw_env.get("HTTP_REFERER") != nullptr;
+    buf << rgw_env.get("REMOTE_ADDR", "") << " - " << s->user->user_id
+    << " [" << s->time << "] \"" << s->info.method 
+    << " " << s->info.request_uri << (!s->info.request_params.empty() ? "?" : "") << s->info.request_params
+    << " HTTP/" << rgw_env.get("HTTP_VERSION", "-") << "\" " << s->err.http_ret << " "
+    << ACCOUNTING_IO(s)->get_bytes_sent() + ACCOUNTING_IO(s)->get_bytes_received() << " "
+    << (referer_hdr ? "\"" : "") << rgw_env.get("HTTP_REFERER", "-") << (referer_hdr ? "\"" : "") << " "
+    << (user_agent_hdr ? "\"" : "") << rgw_env.get("HTTP_USER_AGENT", "-") << (user_agent_hdr ? "\"" : "")
+    << " " << rgw_env.get("HTTP_RANGE", "-");
```

Comment:
within the beast frontend, `handle_connection()` should have access to most of this information between the `parser`, `real_client` and `real_client_io` variables. the only stuff we don't have directly would be the `http_ret` (which we could cache from the call to `rgw::asio::ClientIO::send_status()` and the user_id (which civetweb doesn't show either)